test/format/util.c

```diff
@@ -572,25 +572,45 @@ checkpoint(void *arg)
 }
 
 /*
- * timestamp --
- *     Periodically update the oldest timestamp.
+ * timestamp_once --
+ *     Update the timestamp once.
  */
-WT_THREAD_RET
-timestamp(void *arg)
+void
+timestamp_once(void)
 {
     WT_CONNECTION *conn;
     WT_DECL_RET;
-    WT_SESSION *session;
     char buf[WT_TS_HEX_STRING_SIZE + 64];
-    bool done;
 
-    (void)(arg);
     conn = g.wts_conn;
 
-    testutil_check(conn->open_session(conn, NULL, NULL, &session));
-
     testutil_check(__wt_snprintf(buf, sizeof(buf), "%s", "oldest_timestamp="));
 
+    /*
+     * Lock out transaction timestamp operations. The lock acts as a barrier ensuring we've checked
+     * if the workers have finished, we don't want that line reordered.
+     */
+    testutil_check(pthread_rwlock_wrlock(&g.ts_lock));
+
+    ret = conn->query_timestamp(conn, buf + strlen("oldest_timestamp="), "get=all_committed");
+    testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+    if (ret == 0)
+        testutil_check(conn->set_timestamp(conn, buf));
+
+    testutil_check(pthread_rwlock_unlock(&g.ts_lock));
+}
+
+/*
+ * timestamp --
+ *     Periodically update the oldest timestamp.
+ */
+WT_THREAD_RET
+timestamp(void *arg)
+{
+    bool done;
+
```

Comment:
Nit: `WT_UNUSED`.

Reply:
`format` doesn't use `WT_UNUSED` anywhere -- I can make a pass if you feel strongly.

Reply:
No worries. It was just a nitpick. It's better if it's consistent with the rest of `format`.